server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,675 @@
+'use strict';
+
+var _ = require('lodash');
+var Attendance = require('./attendance.model');
+var mongoose = require('mongoose');
+var ClassYear = require('../classyear/classyear.model');
+var User = require('../user/user.model');
+var SmallGroup = require('../smallgroup/smallgroup.model');
+var config = require('../../config/environment');
+
+function isoDateToTime(isoDate){
+    var date = new Date(isoDate);
+    date.setHours(0,0,0,0);
+    return date.getTime();
+}
+
+// *******************************************************
+// Get list of attendance submissions
+// Restricted to mentors
+// router.get('/', auth.isAuthenticated(), controller.index);
+exports.index = function(req, res) {
+    Attendance.find(function (err, attendance) {
+        if(err) { return handleError(res, err); }
+        return res.json(200, attendance);
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Get a single attendance submission by id
+// Restricted to authenticated users
+// router.get('/:id', auth.isAuthenticated(), controller.show);
+exports.show = function(req, res) {
+    Attendance.findById(req.params.id, function (err, attendance) {
+        if(err) { return handleError(res, err); }
+        if(!attendance) { return res.send(404); }
+        return res.json(attendance);
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Deletes an attendance submission from the DB.
+// Restricted to mentor
+// router.delete('/:id', auth.hasRole('mentor'), controller.destroy);
+exports.destroy = function(req, res) {
+    Attendance.findById(req.params.id, function (err, attendance) {
+        if(err) { return handleError(res, err); }
+        if(!attendance) { return res.send(404); }
+        attendance.remove(function(err) {
+            if(err) { return handleError(res, err); }
+            return res.send(204);
+        });
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Creates a new attendance submission in the DB.
+// DON'T USE : does not generate required data for submission
+// Restricted to admins
+// router.post('/', auth.hasRole('admin'), controller.create);
+exports.create = function(req, res) {
+    Attendance.create(req.body, function(err, attendance) {
+        if(err) { return handleError(res, err); }
+        return res.json(201, attendance);
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Updates an existing attendance submission in the DB.
+// DON'T USE : does not generate required data for submission
+// Restricted to admins
```

Comment:
This was because it was the original creation of the endpoints and a typo. The mere fact that you could at a glance notice that the endpoint should not be for a mentor is proof that it is useful